air/src/execution_step/lambda_applier/applier.rs

```diff
@@ -81,14 +122,35 @@ pub(crate) fn select_from_scalar<'value, 'accessor, 'i>(
         }
     }
 
-    Ok(value)
+    Ok(Cow::Borrowed(value))
+}
+
+fn select_by_functor_from_scalar(value: &JValue, functor: &Functor) -> ExecutionResult<JValue> {
+    match functor {
+        Functor::Length => {
+            let length = value
+                .as_array()
```

Comment:
Length of a map is also a well-defined operation.

Reply:
Didn't get.

Reply:
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.len

Reply:
The reason not to have `length` for object is that `HashMap` for map-type values is not described in a standard, it could be tree of whatever. And imagine that we have this for map-type value why not having it for let's say strings? If so, what about numbers? We could say that length of a number is 1.

And this is the right path to .. JS.

Reply:
Well, having length of string is nice idea. :)

HashMap here is just to give an idea.  Tree has its size too. Yes, we might say that length of a number is 1, but it doesn't mean we should.

Whether we should use same function for vecs and maps is debatable, but if you want separate methods for it, we might rename "length" to something like "vec_length". I'm pretty sure users will need map and string API as well.